docs/source/internals/data_storage.rst

```diff
@@ -1,32 +1,104 @@
-.. todo::
+.. _internal_architecture:data-storage:
 
-    .. _internal_architecture:data-storage:
+************
+Data storage
+************
 
-    ************
-    Data storage
-    ************
+.. todo::
 
     .. _internal_architecture:orm:database:
 
-    Database schema & migrations
-    ============================
+    title: Database schema & migrations
 
     `#4035`_
 
     .. _internal_architecture:orm:repository:
 
-    File repository
-    ===============
+    title: File repository
 
     `#4036`_
 
-    .. _internal_architecture:orm:export:
+.. _internal_architecture:orm:archive:
+
+AiiDA archive format
+====================
+
+An AiiDA archive file is a ``.zip`` or ``.tar.gz`` file with the following content:
+
+* ``metadata.json`` file containing information on the version of AiiDA as well as the database schema.
+* ``data.json`` file containing the nodes and their links.
+* ``nodes/`` directory containing the repository files corresponding to the nodes.
+
+.. _metadata-json:
```

Comment:
You might want to use a longer reference name using the full prefix of where this is - these are global labels. So e.g. something like `_internal_architecture:orm:archive:metadata-json:`